client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx

```diff
@@ -98,22 +99,31 @@ const MarketplaceThankYou = ( {
 		}
 	}, [ transferStatus, areAllProductsFetched, showProgressBar, isJetpack ] );
 
-	// TODO: Use more general steps (not specific to plugins)
-	const steps = useMemo(
-		() =>
-			isJetpack
-				? [ translate( 'Installing plugin' ) ]
-				: [
-						translate( 'Activating the plugin feature' ), // Transferring to Atomic
-						translate( 'Setting up plugin installation' ), // Transferring to Atomic
-						translate( 'Installing plugin' ), // Transferring to Atomic
-						translate( 'Activating plugin' ),
-				  ],
+	const multipleProductTypes = hasMultipleProductTypes( [ pluginSlugs, themeSlugs ] );
+	const additionalSteps = useMarketplaceAdditionalSteps();
+	const randomElement = ( arr: string[] ) => arr[ Math.floor( Math.random() * arr.length ) ];
+
+	const defaultSteps = useMemo(
+		() => {
+			if ( isJetpack ) {
+				return [ randomElement( additionalSteps ) ];
+			}
+
+			return [
+				randomElement( additionalSteps ), // Transferring to Atomic
+				randomElement( additionalSteps ), // Transferring to Atomic
+				randomElement( additionalSteps ), // Transferring to Atomic
+				randomElement( additionalSteps ),
+			];
+		},
 		// We intentionally don't set `isJetpack` as dependency to keep the same steps after the Atomic transfer.
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 		[ translate ]
 	);
-	const additionalSteps = useMarketplaceAdditionalSteps();
+
+	const steps = multipleProductTypes
+		? defaultSteps
+		: [ ...pluginsProgressbarSteps, ...themesProgressbarSteps ];
```

Comment:
For now I don't see this as a problem, but in general I totally agree that this would make it more explicit/clear how we're filling the array. I'll update the code soon to check the arrays when merging them instead of within the hooks.